builder/amazon/ebsvolume/step_snapshot_ebs_volumes.go

```diff
@@ -0,0 +1,167 @@
+package ebsvolume
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	"github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/hashicorp/packer-plugin-sdk/template/interpolate"
+	awscommon "github.com/hashicorp/packer/builder/amazon/common"
+)
+
+type stepSnapshotEBSVolumes struct {
+	PollingConfig *awscommon.AWSPollingConfig
+	AccessConfig  *awscommon.AccessConfig
+	VolumeMapping []BlockDevice
+	//Map of SnapshotID: BlockDevice, Where *BlockDevice is in VolumeMapping
+	snapshotMap map[string]*BlockDevice
+	Ctx         interpolate.Context
+}
+
+func (s *stepSnapshotEBSVolumes) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	ec2conn := state.Get("ec2").(ec2iface.EC2API)
+	instance := state.Get("instance").(*ec2.Instance)
+	ui := state.Get("ui").(packer.Ui)
+
+	s.snapshotMap = make(map[string]*BlockDevice)
+
+	for _, instanceBlockDevice := range instance.BlockDeviceMappings {
```

Comment:
I changed instanceBlockDevices to instanceBlockDevice to make it less confusing that we were looping over a single device at a time